### ISSUE TITLE
Generated images don't correspond to caption

### ISSUE THREAD

**Issue by shenkev** [paarthneekhara/text-to-image#34]
Maybe I'm doing something stupid but essentially I followed your generating images (with pretrained model) steps:

`python generate_thought_vectors.py --caption_file="Data/sample_captions.txt"`

where sample_captions.txt has just these words: "the flower has yellow petals and the center of it is brown" (Is this format correct?)

`python generate_images.py --model_path=Data/Models/latest_model_flowers_temp.ckpt --n_images=8`

A single combined_image_0.jpg comes out in the val_samples folder but the images are DO NOT "have yellow petals and the center of it is brown". As far as I can tell, the model is not conditioning on the text properly. What am I doing wrong?

![image](https://user-images.githubusercontent.com/5405172/31448657-febedb3e-ae72-11e7-8c6a-5e00271d4e8a.png)

**Comment by paarthneekhara** [comment]
Can you try a simple caption like "A red flower" just to check if there is no blunder

**Comment by shenkev** [comment]
This is the output for "A red flower" with n=5:

![image](https://user-images.githubusercontent.com/5405172/31469840-57d0d2fc-aeb1-11e7-89c2-fdf2bcdcea46.png)

**Comment by shenkev** [comment]
Maybe this is relevant?

I replaced ops.py according to the solution mentioned in this issue: https://github.com/paarthneekhara/text-to-image/issues/13

Then I changed sigmoid_cross_entropy_with_logits() to take the "logits" and "labels" arguments explicitly. i.e. sigmoid_cross_entropy_with_logits(logits=... , labels=... ) because it complained about passing arguments explicitly.

**Comment by paarthneekhara** [comment]
Well, there seems something wrong for sure. The output should not look messy like above. Did you train the model or use the pretrained one?

**Comment by shenkev** [comment]
I used the pretrained model. I mean the fact that I'm pointing model_path to latest_model_flowers_temp.ckpt should guarantee that model is loaded. Also the name of the file latest_model_flowers_temp.ckpt indicates this is the up-to-date pretrained model.

`python generate_images.py --model_path=Data/Models/latest_model_flowers_temp.ckpt --n_images=8`

**Comment by paarthneekhara** [comment]
Incase you are still facing troubles, try using a more recent implementation https://github.com/zsdonghao/text-to-image

**Comment by shenkev** [comment]
Thanks, I'll take a look

**Comment by chenxinjing** [comment]
https://github.com/paarthneekharahttps://github.com/shenkev I also encountered the same problem, do you have a solution? I don't modify the parameters. I used tensorflow 1.3.

**Comment by Fuhongshuai** [comment]
I tried using a more recent implementation, but also encountered the same problem, do you have a solution?@shenkev @paarthneekhara

**Comment by silverlilin** [comment]
> Incase you are still facing troubles, try using a more recent implementation https://github.com/zsdonghao/text-to-image

why do i detect the same result? The images obtained from different models are different.
